Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,555 @@
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            q!=amrex::Real(0.0),
+            "Collider-relevant diagnostic does not work for neutral species yet");
+    }
+
+    // function to fill a vector with diags names and create corresponding entry in header
+    std::vector<std::string> all_diag_names;
+    auto add_diag = [&,c=0](
+        const std::string& name, const std::string& header) mutable {
+        m_headers_indices[name] = aux_header_index{header, c++};
+        all_diag_names.push_back(name);
+    };
+
+#if (defined WARPX_DIM_3D)
+    add_diag("dL_dt", "dL_dt(m^-2*s^-1)");
+#elif (defined WARPX_DIM_XZ)
+    add_diag("dL_dt", "dL_dt(m^-1*s^-1)");
+#else
+    add_diag("dL_dt", "dL_dt(s^-1)");
+#endif
+
+    // loop over species
+    for (int i_s = 0; i_s < 2; ++i_s)
+    {
+        // get WarpXParticleContainer class object
+        const WarpXParticleContainer& myspc = mypc.GetParticleContainerFromName(m_beam_name[i_s]);
+
+        if (myspc.DoQED()){
+            add_diag("chimin_"+species_names[i_s], "chi_min_"+species_names[i_s]+"()");
+            add_diag("chiave_"+species_names[i_s], "chi_ave_"+species_names[i_s]+"()");
+            add_diag("chimax_"+species_names[i_s], "chi_max_"+species_names[i_s]+"()");
+        }
+#if (defined WARPX_DIM_3D)
+        add_diag("x_ave_"+species_names[i_s], "x_ave_"+species_names[i_s]+"(m)");
+        add_diag("x_std_"+species_names[i_s], "x_std_"+species_names[i_s]+"(m)");
+        add_diag("y_ave_"+species_names[i_s], "y_ave_"+species_names[i_s]+"(m)");
+        add_diag("y_std_"+species_names[i_s], "y_std_"+species_names[i_s]+"(m)");
+        add_diag("thetax_min_"+species_names[i_s], "theta_x_min_"+species_names[i_s]+"(rad)");
+        add_diag("thetax_ave_"+species_names[i_s], "theta_x_ave_"+species_names[i_s]+"(rad)");
+        add_diag("thetax_max_"+species_names[i_s], "theta_x_max_"+species_names[i_s]+"(rad)");
+        add_diag("thetax_std_"+species_names[i_s], "theta_x_std_"+species_names[i_s]+"(rad)");
+        add_diag("thetay_min_"+species_names[i_s], "theta_y_min_"+species_names[i_s]+"(rad)");
+        add_diag("thetay_ave_"+species_names[i_s], "theta_y_ave_"+species_names[i_s]+"(rad)");
+        add_diag("thetay_max_"+species_names[i_s], "theta_y_max_"+species_names[i_s]+"(rad)");
+        add_diag("thetay_std_"+species_names[i_s], "theta_y_std_"+species_names[i_s]+"(rad)");
+
+#elif (defined WARPX_DIM_XZ)
+        add_diag("x_ave_"+species_names[i_s], "x_ave_"+species_names[i_s]+"(m)");
+        add_diag("x_std_"+species_names[i_s], "x_std_"+species_names[i_s]+"(m)");
+        add_diag("thetax_min_"+species_names[i_s], "theta_x_min_"+species_names[i_s]+"(rad)");
+        add_diag("thetax_ave_"+species_names[i_s], "theta_x_ave_"+species_names[i_s]+"(rad)");
+        add_diag("thetax_max_"+species_names[i_s], "theta_x_max_"+species_names[i_s]+"(rad)");
+        add_diag("thetax_std_"+species_names[i_s], "theta_x_std_"+species_names[i_s]+"(rad)");
```

Comment:
Should we really use `species_names` in this section? A simulation may contain several species and the first two species may not be the two beams we are interested in.

Reply:
Thanks, @lucafedeli88, I think you're right. Now we're using `m_beam_name` instead, which was set by
```cpp
    // read colliding species names - must be 2
    amrex::ParmParse pp_rd_name(m_rd_name);
    pp_rd_name.getarr("species", m_beam_name);
```
and contains correctly only the collider-relevant beam species names.

See [e5d815d](https://github.com/ECP-WarpX/WarpX/pull/4024/commits/e5d815d184aa3238fcddb0cb3d94bbf3066ce64b) for reference. I also switched the order of the particle species in the input file (we start with photons now), to show that this indeed works.